### ISSUE TITLE
"C_Cpp.SwitchHeaderSource" does not respect "files.exclude"

### ISSUE THREAD

**Issue by AdamEttenberger** [microsoft/vscode-cpptools#653]
vscode : 1.11.2
ms-vscode.cpptools: 0.10.5
platform: OSX El Capitan 10.11.6

Steps to reproduce :

1. Create the following folder structure and open it in vscode :
    TESTING
    |-> build
    |-> include
    |-> src
2. modify your global settings to have at least this : "files.exclude": { "build": true }
3. create a main.cpp in src, and a main.h in both include and build
4. (to make it easier to see) write the folder name in each of the files
5. close all files
6. open src/main.cpp
7. Press F4 (or your mapping for "C_Cpp.SwitchHeaderSource")

Notice: It's possible that the build/main.h will be opened rather than the include/main.h

**Comment by AdamEttenberger** [comment]
Possibly related to #541 ?
Some of the symptoms seem similar at least.

**Comment by sean-mcmanus** [comment]
Yeah, we found a bug on Mac in which files.exclude was not being used to exclude which files were parsed (and switched to). I tested it, and it's fixed in the update we're working on releasing, but you might need to delete your browse database (or change the databaseFilename) or update the files.exclude setting in order to get the file removed from the browse database. So interpret the "fixed" label as "a fix will be out soon".
